src/plot_recipes.jl

```diff
@@ -482,252 +271,136 @@ Plot an interval.
-
-    seriestype := :path
-    linecolor   --> color
-    markercolor --> color
-    label --> ""
-    grid --> true
-    markershape --> (add_marker ? :circle : :none)
-
-    for Ii in I
-        @series [Tuple([min(Ii), 0.0]); Tuple([max(Ii), 0.0])]
-    end
+@recipe function plot_interval(I::Interval{N}, ε::N=zero(N)) where {N<:Real}
+    @series LineSegment([min(I), N(0)], [max(I), N(0)]), ε
 end
 
-# ==============================
-# Plot recipe for the empty set
-# ==============================
-
 """
-    plot_emptyset(∅::EmptySet, [ε]; ...)
+    plot_emptyset(∅::EmptySet, [ε]::N=zero(N); ...)
 
 Plot an empty set.
 
 ### Input
 
 - `∅` -- empty set
 - `ε` -- (optional, default: `0`) ignored, used for dispatch
-
-### Output
-
-An empty figure.
 """
 @recipe function plot_emptyset(∅::EmptySet{N}, ε::N=zero(N)) where {N<:Real}
-    label --> ""
-    grid --> true
-    return []
-end
-
-# ===============================
-# Plot recipe for unbounded sets
-# ===============================
-
-"""
-    plot_universe(U::Universe, [ε]; ...)
-
-Plot the universal set.
-
-### Input
-
-- `U` -- universal set
-- `ε` -- (optional, default: `0`) ignored, used for dispatch
-
-### Output
-
-An error, since plotting the universal set is not implemented yet (see #576).
-"""
-@recipe function plot_universe(::Universe{N}, ε::N=zero(N)) where {N<:Real}
-    error("cannot plot the universal set")
-end
-
-"""
-    plot_line(L::Line, [ε]; ...)
-
-Plot a line in 2D.
-
-### Input
-
-- `L` -- line
-- `ε` -- (optional, default: `0`) ignored, used for dispatch
-
-### Output
-
-An error, since plotting a line is not implemented yet (see #576).
-"""
-@recipe function plot_line(::Line{N}, ε::N=zero(N)) where {N<:Real}
-    error("cannot plot an infinite line")
-end
+    label --> DEFAULT_LABEL
+    grid --> DEFAULT_GRID
+    aspect_ratio --> DEFAULT_ASPECT_RATIO
 
-"""
-    plot_halfspace(H::HalfSpace, [ε]; ...)
-
-Plot a half-space in 2D.
-
-### Input
-
-- `H` -- half-space
-- `ε` -- (optional, default: `0`) ignored, used for dispatch
-
-### Output
-
-An error, since plotting a half-space is not implemented yet (see #576).
-"""
-@recipe function plot_halfspace(::HalfSpace{N}, ε::N=zero(N)) where {N<:Real}
-    error("cannot plot a half-space")
+    return []
 end
 
 """
-    plot_hyperplane(H::Hyperplane, [ε]; ...)
+    plot_polyhedron(P::AbstractPolyhedron, [ε]::N=zero(N); ...)
 
-Plot a hyperplane in 2D.
+Plot a polyhedron.
 
 ### Input
 
-- `H` -- hyperplane
+- `P` -- polyhedron
 - `ε` -- (optional, default: `0`) ignored, used for dispatch
 
 ### Output
 
-An error, since plotting a hyperplane is not implemented yet (see #576).
+An error, since plotting unbounded sets is not implemented yet (see
+[#576](https://github.com/JuliaReach/LazySets.jl/issues/576)).
 """
-@recipe function plot_hyperplane(::Hyperplane{N}, ε::N=zero(N)) where {N<:Real}
-    error("cannot plot a hyperplane")
+@recipe function plot_polyhedron(P::AbstractPolyhedron{N}, ε::N=zero(N)
```

Comment:
This choice disables plotting some bounded sets without apparent reason:

```julia
julia> plot(convert(HPolyhedron, rand(Hyperrectangle)))
ERROR: cannot plot an unbounded set
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] apply_recipe(::Dict{Symbol,Any}, ::HPolyhedron{Float64}, ::Float64) at /Users/forets/.julia/packages/RecipesBase/Uz5AO/src/RecipesBase.jl:275
 [3] apply_recipe(::Dict{Symbol,Any}, ::HPolyhedron{Float64}) at /Users/forets/.julia/packages/RecipesBase/Uz5AO/src/RecipesBase.jl:269
 [4] _process_userrecipes(::Plots.Plot{Plots.GRBackend}, ::Dict{Symbol,Any}, ::Tuple{HPolyhedron{Float64}}) at /Users/forets/.julia/dev/Plots/src/pipeline.jl:83
 [5] _plot!(::Plots.Plot{Plots.GRBackend}, ::Dict{Symbol,Any}, ::Tuple{HPolyhedron{Float64}}) at /Users/forets/.julia/dev/Plots/src/plot.jl:178
 [6] #plot#140(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::HPolyhedron{Float64}) at /Users/forets/.julia/dev/Plots/src/plot.jl:57
 [7] plot(::HPolyhedron{Float64}) at /Users/forets/.julia/dev/Plots/src/plot.jl:51
 [8] top-level scope at none:0
```

Possible fix:

```julia
if !isbounded(P)
    error(..)
else
   # call series for general LazySet
end
```

Reply:
I removed the plot recipe for `AbstractPolyhedron`, generalized the plot recipe for `AbstractPolytope` to `AbstractPolyhedron`, and added boundedness checks to the recipes of `LazySet` and `AbstractPolyhedron`.